src/pointer-analysis/dereference.cpp

```diff
@@ -205,6 +208,8 @@ exprt dereferencet::dereference_rec(
   }
 }
 
+/// Attempt to dereference the object at address `expr + offset` and of
+/// type `type`. Throws an exception if unsuccessful.
```

Comment:
This would be helpful if it explained a bit more what the expected behaviour is (I can figure it out from reading the code, but would be helpful to have that in the docs too). e.g. what does it mean when expr is an if_exprt, and what is the expected return value?